libbeat/idxmgmt/idxmgmt.go

```diff
@@ -99,30 +100,60 @@ func DefaultSupport(log *logp.Logger, info beat.Info, configRoot *config.C) (Sup
 
 // MakeDefaultSupport creates some default index management support, with a
 // custom ILM support implementation.
-func MakeDefaultSupport(ilmSupport ilm.SupportFactory) SupportFactory {
+func MakeDefaultSupport(ilmSupport lifecycle.SupportFactory) SupportFactory {
 	if ilmSupport == nil {
-		ilmSupport = ilm.DefaultSupport
+		ilmSupport = lifecycle.DefaultSupport
 	}
 
 	return func(log *logp.Logger, info beat.Info, configRoot *config.C) (Supporter, error) {
 		const logName = "index-management"
+		if log == nil {
+			log = logp.NewLogger(logName)
+		} else {
+			log = log.Named(logName)
+		}
 
-		cfg := struct {
-			ILM       *config.C        `config:"setup.ilm"`
-			Template  *config.C        `config:"setup.template"`
-			Output    config.Namespace `config:"output"`
-			Migration *config.C        `config:"migration.6_to_7"`
+		// first fetch the ES output, check if we're running against serverless, use that to set a default config.
+		// the Supporter only cares about lifecycle config for checking if ILM/DSL is enabled or disabled.
+		outCfg := struct {
+			Output config.Namespace `config:"output"`
 		}{}
 		if configRoot != nil {
-			if err := configRoot.Unpack(&cfg); err != nil {
-				return nil, err
+			err := configRoot.Unpack(&outCfg)
+			if err != nil {
+				return nil, fmt.Errorf("error unpacking output config while making index support: %w", err)
 			}
 		}
 
-		if log == nil {
-			log = logp.NewLogger(logName)
-		} else {
-			log = log.Named(logName)
+		defaultLifecycle := lifecycle.DefaultILMConfig(info)
+		if outCfg.Output.IsSet() && outCfg.Output.Name() == "elasticsearch" {
+			esClient, err := eslegclient.NewConnectedClient(outCfg.Output.Config(), info.Beat)
+			if err != nil {
+				// this is for the benefit of tests as well as various retry mechanisms;
+				// if we can't connect to ES, print a warning and fallback to ILM, tests won't care and non-test environments can
+				// retry the connection
+				logp.L().Warnf("could not connect to ES to determine ES type during index setup. Will default to ILM.")
```

Comment:
Should this include the error for context?

```suggestion
				logp.L().Warnw("could not connect to ES to determine ES type during index setup. Will default to ILM.", "error", err)
```